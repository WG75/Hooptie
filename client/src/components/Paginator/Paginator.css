.paginator {
    display: flex;
}

.paginator__item {
    margin: 12px;
    font-size: 18px;
}