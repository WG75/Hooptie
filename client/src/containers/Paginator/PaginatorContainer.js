// @flow

import { connect } from 'react-redux';
import qs from 'query-string';
import PaginatorComponent from '../../components/Paginator';
import { fetchCars } from '../CarsFilter/CarsFilterRedux';


function mapStateToProps(state: Object) {
  return {
    nextPage: state.filter.nextPage,
    prevPage: state.filter.prevPage,
    lastPage: state.filter.lastPage,
    currentPage: state.filter.currentPage,
    query: qs.parse(state.router.location.search),
  };
}

function mapDispatchToProps(dispatch: (() => any) => any) {
  return {
    onChange: (query: Object) => dispatch(fetchCars(query)),
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(PaginatorComponent);
