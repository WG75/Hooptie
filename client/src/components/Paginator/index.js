// @flow

import React from 'react';
import { Link } from 'react-router-dom';
import './Paginator.css';

type Props = {
  nextPage: number,
  prevPage: number,
  lastPage: number,
  currentPage: number,
  onChange: (page: {page: number}) => void,
  query: Object
};

const Paginator = ({
  nextPage,
  prevPage,
  lastPage,
  currentPage,
  onChange,
  query,
}: Props) => (
  <ul className="paginator">
    <li className="paginator__item">
      <a className="link" onClick={() => onChange({ ...query, page: 1 })}>
        First
      </a>
    </li>
    <li className="paginator__item">
      <a className="link" onClick={() => onChange({ ...query, page: prevPage })}>
        Previuos
      </a>
    </li>
    <li className="paginator__item">
      <span>{`Page ${currentPage} of ${lastPage}`}</span>
    </li>
    <li className="paginator__item">
      <a className="link" onClick={() => onChange({ ...query, page: nextPage })}>
        Next
      </a>
    </li>
    <li className="paginator__item">
      <a className="link" onClick={() => onChange({ ...query, page: lastPage })}>
        Last
      </a>
    </li>
  </ul>
);

export default Paginator;
