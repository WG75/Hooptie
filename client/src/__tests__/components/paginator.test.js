import React from 'react';
import { create } from 'react-test-renderer';
import Enzyme from '../../config/enzyme';
import Paginator from '../../components/Paginator';

describe('Paginator component', () => {
  const onChangeMock = jest.fn();

  const wrapper = Enzyme.shallow(
    <Paginator
      nextPage={3}
      prevPage={1}
      lastPage={10}
      currentPage={2}
      onChange={onChangeMock}
      query={{
        page: 2,
        sort: 'dec',
        color: 'red',
        manufacturer: 'BMW',
      }
      }
    />,
  );

  it('call onChange handler giving it the next page as params', () => {
    wrapper.find('.link').at(0).simulate('click');
    expect(onChangeMock.mock.calls.length).toBe(1);
    expect(onChangeMock.mock.calls[0][0]).toEqual({
      page: 1,
      sort: 'dec',
      color: 'red',
      manufacturer: 'BMW',
    });
  });


  test('it matches the snapshot', () => {
    const component = create(<Paginator
      nextPage={3}
      prevPage={1}
      lastPage={10}
      currentPage={2}
      onChange={onChangeMock}
      query={{
        page: 2,
        sort: 'dec',
        color: 'red',
        manufacturer: 'BMW',
      }
        }
    />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
