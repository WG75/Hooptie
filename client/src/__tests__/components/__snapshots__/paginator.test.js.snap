// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator component it matches the snapshot 1`] = `
<ul
  className="paginator"
>
  <li
    className="paginator__item"
  >
    <a
      className="link"
      onClick={[Function]}
    >
      First
    </a>
  </li>
  <li
    className="paginator__item"
  >
    <a
      className="link"
      onClick={[Function]}
    >
      Previuos
    </a>
  </li>
  <li
    className="paginator__item"
  >
    <span>
      Page 2 of 10
    </span>
  </li>
  <li
    className="paginator__item"
  >
    <a
      className="link"
      onClick={[Function]}
    >
      Next
    </a>
  </li>
  <li
    className="paginator__item"
  >
    <a
      className="link"
      onClick={[Function]}
    >
      Last
    </a>
  </li>
</ul>
`;

exports[`Select component it matches the snapshot 1`] = `
<div
  className="select-container select"
>
  <span
    className="select-container__label"
  >
    difficulty
  </span>
  <div
    className="react-select-container css-2b097c-container"
    onKeyDown={[Function]}
  >
    <div
      className="react-select__control css-yk16xz-control"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="react-select__value-container react-select__value-container--has-value css-1hwfws3"
      >
        <div
          className="react-select__single-value css-1uccc91-singleValue"
        >
          easy
        </div>
        <input
          className="css-62g3xt-dummyInput"
          disabled={false}
          id="react-select-2-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          readOnly={true}
          tabIndex="0"
          value=""
        />
      </div>
      <div
        className="react-select__indicators css-1wy0on6"
      >
        <span
          className="react-select__indicator-separator css-1okebmr-indicatorSeparator"
        />
        <div
          aria-hidden="true"
          className="react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-19bqh2r"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
